src/json.c

```diff
@@ -159,6 +159,70 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type, QueryError *stat
 //   return ret;
 // }
 
+#define BIT_CAST(type, value) ({ \
+  __typeof__(value) value_var = (value); \
+  static_assert(sizeof(type) == sizeof(value_var), "type and value must have the same size"); \
+  const void *ptr = &(value_var); \
+  *(type *)ptr; \
+})
+
+// Rounding the input to nearest even (float with 16 trailing zeros),
+// and returning the 16 significant bits of the float as uint16_t.
+// Calculation inspired by:
+// https://gitlab.com/libeigen/eigen/-/blob/d626762e3ff6cdcdf65325e6edf27c995029786d/Eigen/src/Core/arch/Default/BFloat16.h#L403
+static inline uint16_t floatToBF16bits(float input) {
```

Comment:
Consider using `typedef` for float16 and bfloat16 as we did in the lib

Reply:
I think it will only add confusion here.
We only care about the bits in this file (for any callback `getJSONElementFunc VecSimGetJSONCallback(VecSimType type)` returns), the type itself is not checked outside this function scope, and it's more clear that this function returns `BF16bits` and not a type that we cannot actually use for anything else (casts, value reading)